backend/cmm.ml

```diff
@@ -319,14 +319,10 @@ let iter_shallow_tail f = function
       f e1;
       f e2;
       true
-  | Cregion e ->
-      f e;
-      true
-  | Ctail e ->
-      f e;
-      true
   | Cexit _ | Cop (Craise _, _, _) ->
       true
+  | Cregion _
+  | Ctail _
```

Comment:
Shouldn't `Ctail` still count as tail?

Potentially we could also treat e.g. `e` in `Cregion (let x = ... in Ctail(e))` as being tail. That might come up quite often in the output of flambda after Luke's patch.

Reply:
That would indeed be correct but I think it would also be trickier and relatively low value, so I'd like to leave it out of this fix.